lib/kino.ex

```diff
@@ -445,6 +445,32 @@ defmodule Kino do
     DynamicSupervisor.start_child(Kino.DynamicSupervisor, child_spec)
   end
 
+  @doc """
+  Similar to `start_child/2` but returns the new pid or raises an error.
+  """
+  @spec start_child!(Supervisor.child_spec() | {module(), term()} | module()) :: pid()
+  def start_child!(child_spec) do
+    case start_child(child_spec) do
+      {:ok, pid} ->
+        pid
+
+      {:ok, pid, _info} ->
+        pid
+
+      {:error, reason} ->
+        raise "failed to start child with the spec #{Kernel.inspect(child_spec)}.\n" <>
+                "Reason: #{start_supervised_error(reason)}"
+    end
+  end
+
+  defp start_supervised_error({{:EXIT, reason}, info}) when is_tuple(info),
+    do: Exception.format_exit(reason)
+
+  defp start_supervised_error({reason, info}) when is_tuple(info),
+    do: Exception.format_exit(reason)
+
+  defp start_supervised_error(reason), do: Exception.format_exit(reason)
```

Comment:
ExUnit does `{:start_spec, reason}`, but I think it is confusing since it doesn't really have to do with the child spec itself:

```
** (RuntimeError) failed to start child with the spec Foo.
Reason: bad child specification, got: {:already_started, #PID<0.330.0>}
```

Reply:
Your call if you want to change it here. :)